### ISSUE TITLE
error "non-local lambda expression cannot have a capture-default" when make

### ISSUE THREAD

**Issue by ligewu** [Reconfigurable-Computing-CalPoly-Pomona/Reconfigurable-BLAST-N#1]
g++ -std=c++17 -Iinclude/parallel-hashmap/ -Wall -Wextra -march=native -O3 -lm -lpthread -fopenmp -pipe   -c -o lib/blastn.o lib/blastn.cpp
In file included from lib/../util/types.hpp:10,
                 from lib/dust.hpp:3,
                 from lib/blastn.hpp:7,
                 from lib/blastn.cpp:1:
lib/../util/constants.hpp:23:19: error: non-local lambda expression cannot have a capture-default
   23 | auto pack_find = [**&**](char letter) {
      |                   ^
vi make: *** [<builtin>: lib/blastn.o] Error 1

**Comment by JacobLondon** [comment]
A few things: I could not recreate this issue myself, I have a suspicion that your G++ isn't 7.4+ or is aliasing clang++ or some other compiler, it looks like there is a GCC extension which allows this to compile, but clang denies it. If this isn't the issue, then go to line 23 in cpp/util/constants.hpp and replace the 'auto pack_find = [&](char letter) {' with 'auto pack_find = [](char letter) {' to not capture anything (delete the ampersand &).

Another note, the build has only even been done on Debian based distros and Windows with VC++, Mac/BSD is untested but may work.

In summary: let me know if the issue persists after make sure your:
1. Compiler is indeed G++ 7.4+
    $ g++ --version
2. Replace line 23 '&' with ''

If the issue persists after this, I'll have to make the lambda into an inline function.

**Comment by ligewu** [comment]
failed on Fedora 33, but successed on Ubuntu, I'll use the later one.
